src/dra/5298C.c

```diff
@@ -2,7 +2,728 @@
+                } else {
+                    func_801027C4(1);
+                case 2:
+                    D_800978F8++;
+                }
+                break;
+            case 1:
+                if (func_801025F4() != 0) {
+                    D_800978F8++;
+                }
+                break;
+            case 3:
+                if (g_IsUsingCd) {
+                    if (D_80097C98 == 2) {
+                        func_800EA5AC(0xFF, 0xFF, 0xFF, 0xFF);
+                    }
+                    break;
+                } else if (D_80097928 != 0) {
+                    D_80097910 = D_800A3C58[g_StageId * 11];
+                    if (D_800973FC != 1) {
+                        PlaySfx(0xA);
+                        if (func_80131F68() == false) {
+                            PlaySfx(D_80097910);
+                            D_80097928 = 0;
+                        } else {
+                            break;
+                        }
+                    }
+                }
+                if (D_80137598 != 0) {
+                    func_8010427C();
+                    D_80137598 = 0;
+                }
+                layer = g_api.o.tileLayers[D_801375BC->tileLayoutId].fg;
+                if (layer->unkB & 0x80) {
+                    D_801375BC = &g_api.o.rooms[((layer->unkB & 0x7F))].load;
+                    // do-while prevents instruction reordering
+                    do {
+                        D_8009791C = layer->zPriority;
+                    } while (0);
+                    D_80097920 = layer->unkE;
+                } else {
+                    D_8009791C = (playerX >> 8) + g_CurrentRoom.left;
+                    D_80097920 = (playerY >> 8) + g_CurrentRoom.top;
+                }
+                D_8013759C = PLAYER.posX.val;
+                D_801375A0 = PLAYER.posY.val;
+                PLAYER.posX.i.hi = D_801375C0 + g_Camera.posX.i.hi;
+                PLAYER.posY.i.hi = D_801375C4 + g_Camera.posY.i.hi;
+                if (D_8003C708.flags & 0x60) {
+                    func_800EAF28(1);
+                }
+                func_800EA538(2);
+                if (D_801375BC->tilesetId != 0) {
+                    func_800EA5E4((D_801375BC->tilesetId + 0x7FFF) | 0x4000);
+                }
+                if (D_801375BC->objGfxId != 0) {
+                    func_800EAF28(D_801375BC->objGfxId + 0x7FFF);
+                    D_80097904 = D_801375BC->objGfxId + 0x7FFF;
+                } else {
+                    D_80097904 = 0;
+                }
+                func_800F2404(1);
+                PLAYER.posY.i.hi = (u8)PLAYER.posY.i.hi;
+                PLAYER.posX.i.hi = (u8)PLAYER.posX.i.hi;
+                LoadRoomLayer(D_801375BC->tileLayoutId);
+                if (D_8003C708.flags & 0x20) {
+                    func_800EAF28(3);
+                }
+                if (D_8003C708.flags & 0x40) {
+                    func_800EAF28(4);
+                }
+                g_CurrentRoom.x =
```

Comment:
I'm wondering if this is actually a RoomHeader and this is accessing left/top/right/bottom. https://github.com/Xeeynamo/sotn-decomp/blob/b9163ae1f80d697ae7e42656f7ea8c741ef1e48f/include/game.h#L493
Subtracting IDs from the dimensions doesn't seem to make much sense. It may also be that RoomHeader is actually two structs and the one here is

```
    /* 0x0 */ u8 left;
    /* 0x1 */ u8 top;
    /* 0x2 */ u8 right;
    /* 0x3 */ u8 bottom;
```